src/build.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const core = require('@actions/core')
+const { exec } = require('@actions/exec')
+const parseReleaseMetadata = require('./utils/parseReleaseMetadata')
+
+function getMonorepoData({ context, inputs }) {
+  if (context?.payload?.pull_request) {
+    return parseReleaseMetadata(context.payload.pull_request)
+  }
+
+  return {
+    monorepoPackage: inputs['monorepo-package'],
+    monorepoRoot: inputs['monorepo-root'],
+  }
+}
+
+module.exports = async function ({ context, inputs }) {
+  const { monorepoPackage, monorepoRoot } = getMonorepoData({ context, inputs })
+
+  const buildCommands = inputs['build-command']
```

Comment:
I see what you're doing here. It looks reasonable, but probably not very robust. 

Very simple example: what if the user puts an additional new line in the command? I haven't tried but I suspect it'll try to execute an empty command, thereby failing. This one may be easy to fix, but I wonder what other cases we may come across. Again this is because we're pretending to be a shell, which we aren't

Reply:
Yes, I just fixed the case you mentioned, but I'm really not sure about how to avoid doing this in javascript. I tried one more time to go back to bash but we do need to run it when closing the PR to publish the npm package. 

One thing we can do is instead of using the release body to store metadata we just save it to a .json file and use [upload-artifact](https://github.com/actions/upload-artifact) and [download-artifact](https://github.com/actions/download-artifact) actions to store and retrieve it between runs. This would not be a simple change tho so I think it should be done in another PR if you think that's a good idea.

Reply:
no need to go that far as using artifacts